### PR TITLE
[SYCLomatic] Add helper function for debug features.

### PR DESCRIPTION
**Pull request by ShengchenJ**

### CODE REVIEW

File: clang/runtime/dpct-rt/include/debug/schema.hpp [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+      std::make_shared<Schema>(VarName, TypeName, type_size, SchemaTy,
+                               IsBasicType, ValTy, ValSize, Location));
+}
+
+inline std::shared_ptr<Schema> get_type_schema(dpct_json::value &v) {
+  auto obj = v.get_value<dpct_json::object>();
+  std::string schema_name = obj.get("TypeName").get_value<std::string>();
+
+  std::string schema_type_name =
+      obj.get("schema_type").get_value<std::string>();
```
Comment (wenhuiNi) [inline clang/runtime/dpct-rt/include/debug/schema.hpp]: obj.get("schema_type")， I think we need unify the style of schema keyword.
"TypeName" "FieldNum" "IsVirtual", most keyword is named as CamelCase style, only this one is slash.
I prefer "SchemaType"

File: clang/runtime/dpct-rt/include/debug/schema.hpp [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+}
+
+inline std::shared_ptr<Schema> get_data_schema(dpct_json::value &v) {
+  dpct_json::object data_obj = v.get_value<dpct_json::object>();
+  std::string var_name = data_obj.get("VarName").get_value<std::string>();
+  std::string type_name = data_obj.get("TypeName").get_value<std::string>();
+  const schema_type local_type =
+      get_schema_type(data_obj.get("SchemaType").get_value<std::string>());
+  bool is_basic_type = data_obj.get("IsBasicType").get_value<bool>();
+  size_t type_size = data_obj.get("TypeSize").get_value<int>();
```
Comment (wenhuiNi) [inline clang/runtime/dpct-rt/include/debug/schema.hpp:217]: Not TypeSize, should be VarSIze

File: clang/runtime/dpct-rt/include/debug/schema.hpp [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+  std::string val_type = data_obj.get("ValType").get_value<std::string>();
+  size_t val_size = data_obj.get("ValSize").get_value<int>();
```
Comment (wenhuiNi) [inline clang/runtime/dpct-rt/include/debug/schema.hpp:219]: What do TypeSize and ValSize represent？ I think TypeSizeshould be stored in TypeSchema

File: clang/runtime/dpct-rt/include/debug/schema.hpp [hunk 4]
```diff
@@ -0,0 +1,391 @@
...
+  std::string location = data_obj.get("Location").get_value<std::string>();
```
Comment (wenhuiNi) [inline clang/runtime/dpct-rt/include/debug/schema.hpp:216]: VarSchema would comtain the msg about where the var is defined. I think it should be get from jsonobj

File: clang/runtime/dpct-rt/include/debug/schema.hpp (hunk 1 above)
Comment (ShengchenJ) [reply clang/runtime/dpct-rt/include/debug/schema.hpp]: Updated

File: clang/runtime/dpct-rt/include/debug/schema.hpp (hunk 4 above)
Comment (ShengchenJ) [reply clang/runtime/dpct-rt/include/debug/schema.hpp:216]: After syncing with wenhui, plan to remove the FilePath json item. And using the API UID to keep the details.

**Review by AndyCHHuang — APPROVED** [review]

**Pull request merged**